### ISSUE TITLE
IPv6 support

### ISSUE THREAD

**Issue by piscisaureus** [joyent/libuv#19]
The purpose of this issue is mainly to annoy Ryan.
Besides, we _do_ need IPv6 support.

**Comment by ry** [comment]
sure - it's easy to add. let's put it off until we have to fix the ::1 test in node.

**Comment by ry** [comment]
want

``` c
int uv_bind6(uv_tcp_t*, struct sockaddr_in6);
struct sockaddr_in6 uv_ip6_addr(const char* host, int port);
```

**Comment by mattstevens** [comment]
I added IPv6 bind to fix a Mobile Safari issue (iOS waits on an IPv6 timeout for every connection if present) but ran into a snag I could use some help with.

XP doesn't have inet_pton so instead of a nice simple uv_ip6_addr it looks like we'll have to use getaddrinfo. Is there a better XP-compatible option? If not, should uv_ip6_addr be platform-specific?

Here's what I have so far: https://github.com/mattstevens/libuv/compare/master...ipv6

**Comment by bnoordhuis** [comment]
@mattstevens: does [this](https://gist.github.com/1048609) work on Windows XP?

**Comment by piscisaureus** [comment]
c-ares provides an implementation of inet_pton that works on xp, although it is not part of the public API. We could use that however. Alternatively we could fall back to getaddrinfo on xp - I prefer to use inet_pton on platforms where it is available.

**Comment by piscisaureus** [comment]
We can also choose to just not support ipv6 on xp. XP does not support IPv6 by default. An experimental ipv6 driver is available, but afaik noone really uses that. 

Anyway, uv_ip6_addr needs top be platform-specific.

**Comment by bnoordhuis** [comment]
The c-ares option sounds like a good idea, though. Cross-platform and less code to maintain.

**Comment by mattstevens** [comment]
Thanks for the help! Looks like XP also has [WSAStringToAddress](http://msdn.microsoft.com/en-us/library/ms742214.aspx). I'll see if I can get one of these working.

**Comment by ry** [comment]
We've got c-ares compiled in we might as well use that!
